internal/compiler/passes/binding_analysis.rs

```diff
@@ -139,6 +139,12 @@ fn perform_binding_analysis(
         &(),
         &mut |e, _| analyze_element(e, &mut context, reverse_aliases, diag),
     );
+
+    component.init_code.borrow().iter().for_each(|init_expr| {
```

Comment:
It doesn't feel right to have a visit of the init_code there in this file.
Why is this required for the init_code and not for other stuff? (root_constraint, popup x/y) maybe that's missing as well?

Reply:
AFAICS the popup is visisted via recurse_elem() above - as that's an element by itself. The root_constraints has no expressions AFAICS. How would they be integrated in this binding analysis function?

Reply:
i was thinking of the x and y NamedReference.
Anyway, this is probably unrelated anyway.